src/crimson/osd/object_context_loader.cc

```diff
@@ -47,27 +47,32 @@ using crimson::common::local_conf;
           crimson::ct_error::enoent::make()
         };
       }
+      auto resolved_oid = resolve_oid(head->get_head_ss(), oid);
+      if (!resolved_oid) {
+        ERRORDPP("clone {} not found", dpp, oid);
+        return load_obc_iertr::future<>{
+          crimson::ct_error::enoent::make()
+        };
+      }
+      if (resolved_oid->is_head()) {
+        // See resolve_oid
+        return std::move(func)(head, head);
+      }
       return this->with_clone_obc_only<State>(std::move(head),
-                                              oid,
+                                              *resolved_oid,
                                               std::move(func));
     });
   }
 
   template<RWState::State State>
   ObjectContextLoader::load_obc_iertr::future<>
   ObjectContextLoader::with_clone_obc_only(ObjectContextRef head,
-                                           hobject_t oid,
+                                           hobject_t clone_oid,
```

Comment:
Should we assert that `clone_oid` is actually a clone one so that future development won't violate this rule?

Reply:
will be addressed in the (blocked-by) PR and rebased